frontend/pages/[lng]/courses/[id]/index.tsx

```diff
@@ -22,15 +22,11 @@ export const CourseDetailsFromSlugQuery = gql`
 
 interface CourseProps {
   admin: boolean
-  router: SingletonRouter
 }
 const Course = (props: CourseProps) => {
-  const { admin, router } = props
+  const { admin } = props
 
-  const slug =
-    router?.query?.id && typeof router.query.id === "string"
-      ? router.query.id
-      : ""
+  const slug = useQueryParameter("id") ? useQueryParameter("id") : ""
```

Comment:
Could use nullish coalescing here (and also in other places in this pr) instead of the ternary operator. https://devblogs.microsoft.com/typescript/announcing-typescript-3-7/#nullish-coalescing